processing/src/main/java/io/druid/query/aggregation/post/JavaScriptPostAggregator.java

```diff
@@ -86,7 +86,8 @@ public double apply(Object[] args)
   private final List<String> fieldNames;
   private final String function;
 
-  private final Function fn;
+  // This variable is lazily initialized to avoid unnecessary JavaScript compilation during JSON serde
+  private Function fn;
```

Comment:
Similar thread-safety questions to the other files.

Reply:
Thanks. Fixed.